### PR TITLE
Add the option to ignore actions from other ES clients

### PR DESCRIPTION
**Pull request by tburgin**
* Also bump version to 1.15

### CODE REVIEW

File: Source/common/SNTConfigurator.h [hunk 1]
```diff
@@ -320,6 +320,11 @@
 ///
 @property(readonly, nonatomic) BOOL enableForkAndExitLogging;
 
+///
+///  If true, skip processing any actions from other endpoint security clients. Defaults to false.
```
Comment (russellhancox) [inline Source/common/SNTConfigurator.h:324]: Call out that this only applies when running as a sysx; _other_ implies it but let's make it clear.

File: Source/common/SNTConfigurator.m [hunk 2]
```diff
@@ -79,6 +79,7 @@ @implementation SNTConfigurator
 static NSString *const kEnableSystemExtension = @"EnableSystemExtension";
 
 static NSString *const kEnableForkAndExitLogging = @"EnableForkAndExitLogging";
+static NSString *const kSkipOtherEndpointSecurityClients = @"SkipOtherEndpointSecurityClients";
```
Comment (russellhancox) [inline Source/common/SNTConfigurator.m:82]: _Skip_ sounds like we might be preventing them from doing processing. Maybe _ignore_?

File: Source/santad/EventProviders/SNTEndpointSecurityManager.mm [hunk 3]
```diff
@@ -77,6 +77,14 @@ - (void)establishClient API_AVAILABLE(macos(10.15)) {
 
     es_client_t *client = NULL;
     es_new_client_result_t ret = es_new_client(&client, ^(es_client_t *c, const es_message_t *m) {
+      // If enabled, skip any action generated from another endpoint security client.
+      if (config.skipOtherEndpointSecurityClients && m->process->is_es_client) {
```
Comment (russellhancox) [inline Source/santad/EventProviders/SNTEndpointSecurityManager.mm:81]: Reverse these conditions - I think `is_es_client` is going to be very rare and is a fairly simple field access vs a property get (with all the KVO goodness that comes with it).

**Review by russellhancox — CHANGES_REQUESTED** [review]

File: Source/common/SNTConfigurator.h (hunk 1 above)
Comment (tburgin) [reply Source/common/SNTConfigurator.h]: Done

File: Source/common/SNTConfigurator.m (hunk 2 above)
Comment (tburgin) [reply Source/common/SNTConfigurator.m]: Good idea, done

File: Source/santad/EventProviders/SNTEndpointSecurityManager.mm (hunk 3 above)
Comment (tburgin) [reply Source/santad/EventProviders/SNTEndpointSecurityManager.mm]: Done

**Review by russellhancox — APPROVED** [review]

**Pull request merged**